File: src/brs/web/api/http/handler/GetAccountPublicKey.java
package brs.web.api.http.handler;

import brs.Account;
import brs.SignumException;
import brs.services.ParameterService;
import brs.util.Convert;
import brs.util.JSON;
import brs.web.api.http.ApiServlet;
import brs.web.api.http.common.LegacyDocTag;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;

import javax.servlet.http.HttpServletRequest;

import static brs.web.api.http.common.Parameters.ACCOUNT_PARAMETER;
import static brs.web.api.http.common.ResultFields.PUBLIC_KEY_RESPONSE;

public final class GetAccountPublicKey extends ApiServlet.JsonRequestHandler {

  private final ParameterService parameterService;

  public GetAccountPublicKey(ParameterService parameterService) {
    super(new LegacyDocTag[] {LegacyDocTag.ACCOUNTS}, ACCOUNT_PARAMETER);
    this.parameterService = parameterService;
  }

  @Override
  protected
  JsonElement processRequest(HttpServletRequest req) throws SignumException {

    Account account = parameterService.getAccount(req);

    if (account.getPublicKey() != null) {
      JsonObject response = new JsonObject();
      response.addProperty(PUBLIC_KEY_RESPONSE, Convert.toHexString(account.getPublicKey()));
      return response;
    } else {
      return JSON.emptyJSON;
    }
  }

}

File: src/main/java/brs/http/GetAccountPublicKey.kt
package brs.http

import brs.BurstException
import brs.http.common.Parameters.ACCOUNT_PARAMETER
import brs.http.common.ResultFields.PUBLIC_KEY_RESPONSE
import brs.services.ParameterService
import brs.util.JSON
import brs.util.toHexString
import com.google.gson.JsonElement
import com.google.gson.JsonObject
import javax.servlet.http.HttpServletRequest

internal class GetAccountPublicKey internal constructor(private val parameterService: ParameterService) : APIServlet.JsonRequestHandler(arrayOf(APITag.ACCOUNTS), ACCOUNT_PARAMETER) {

    override suspend fun processRequest(request: HttpServletRequest): JsonElement {

        val account = parameterService.getAccount(request) ?: return JSONResponses.INCORRECT_ACCOUNT

        return if (account.publicKey != null) {
            val response = JsonObject()
            response.addProperty(PUBLIC_KEY_RESPONSE, account.publicKey!!.toHexString())
            response
        } else {
            JSON.emptyJSON
        }
    }
}
